File: src/main/java/com/my/lostfound/service/ItemService.java
package com.my.lostfound.service;



import com.my.lostfound.dto.ItemRequestDto;
import com.my.lostfound.dto.ItemResponseDto;
import org.springframework.data.domain.Page;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

public interface ItemService {

    // Create
    ItemResponseDto createItem(ItemRequestDto dto);

    // Get All (Pagination + Sorting)
    Page<ItemResponseDto> getAllItems(
            int page,
            int size,
            String sortBy,
            String direction
    );

    // Get By ID
    ItemResponseDto getItemById(Long id);

    // Update
    ItemResponseDto updateItem(Long id, ItemRequestDto dto);

    // Soft Delete
    void deleteItem(Long id);

    // Search
    List<ItemResponseDto> searchByTitle(String keyword);

    // Filter
    List<ItemResponseDto> filterItems(String location, boolean found);

    // File Upload
    String uploadImage(Long id, MultipartFile file);

}

File: Frontend/src/services/itemService.js
import axios from '../api/axiosConfig';

const ItemService = {

    getAllItems: async (page = 0, size = 10) => {
        return await axios.get(`/items?page=${page}&size=${size}`);
    },

    createItem: async (itemData) => {
        return await axios.post('/items', itemData);
    },

    uploadImage: async (id, file) => {
        const formData = new FormData();
        formData.append('file', file);
        
        return await axios.post(`/items/${id}/upload`, formData, {
            headers: {
                'Content-Type': 'multipart/form-data'
            }
        });
    },

    getItemById: async (id) => {
        return await axios.get(`/items/${id}`);
    },

    deleteItem: async (id) => {
        return await axios.delete(`/items/${id}`);
    },

    updateItem: async (id, itemData) => {
        return await axios.put(`/items/${id}`, itemData);
    },

    searchItems: async (keyword) => {
        return await axios.get(`/items/search?keyword=${keyword}`);
    }
};

export default ItemService;